cmd/osbuild-worker/main.go

```diff
@@ -188,6 +197,39 @@ func main() {
 		logrus.Fatalf("Could not load config file '%s': %v", configFile, err)
 	}
 
+	logrus.SetReportCaller(true)
+	logrus.AddHook(&common.BuildHook{})
+	logrus.SetOutput(os.Stdout)
+	logLevel, err := logrus.ParseLevel(config.Logging.Level)
+	if err == nil {
+		logrus.SetLevel(logLevel)
+	} else {
+		logrus.Info("Failed to load loglevel from config:", err)
+	}
+
+	// logger configuration
+	switch config.Logging.Format {
+	case "journal", "":
+		// If we are running under systemd, use the journal. Otherwise,
+		// fallback to text formatter.
+		if journal.Enabled() {
+			logrus.Info("Switching to journal logging mode, use logging.no_discard to keep writing to standard output/error")
+			logrus.SetFormatter(&logrus.JSONFormatter{})
+			logrus.AddHook(&common.JournalHook{})
+			if !config.Logging.NoDiscard {
+				logrus.SetOutput(io.Discard)
+			}
+		} else {
+			logrus.SetFormatter(&logrus.TextFormatter{})
+		}
+	case "text":
+		logrus.SetFormatter(&logrus.TextFormatter{})
+	case "json":
+		logrus.SetFormatter(&logrus.JSONFormatter{})
+	default:
+		logrus.Infof("Failed to set logging format from config, '%s' is not a valid option", config.Logging.Format)
+	}
+
```

Comment:
FYI this was copy-pasted from `composer/main.go` we already use it there.

Reply:
Would it make sense to have some common logging setup function to actually benefit from having both composer and worker in the same repo?

Reply:
I would say no, these are completely different binaries, separated by design. Little bit of copying is sometimes better than a dependency.